packages/postgres/lib/postgres_p.js

```diff
@@ -26,65 +26,100 @@ module.exports = function capturePostgres(pg) {
     return pg;
 
   pg.Client.prototype.__query = pg.Client.prototype.query;
+  pg.Client.prototype.query = captureQuery;
+  return pg;
+};
 
-  pg.Client.prototype.query = function captureQuery(sql, values, callback, segment) {
-    var parent = AWSXRay.resolveSegment(segment);
-    delete arguments[3];
+function resolveArguments(argsObj) {
+  var args = {};
 
-    if (!parent) {
-      AWSXRay.getLogger().info('Failed to capture Postgres. Cannot resolve sub/segment.');
-      return this.__query.apply(this, arguments);
+  if (argsObj && argsObj.length > 0) {
+    if (argsObj[0] instanceof Object) {
+      args.sql = argsObj[0].text;
+      args.values = argsObj[0].values;
+      args.callback = argsObj[1] || argsObj[0].submit;
+    } else {
+      args.sql = argsObj[0];
+      args.values = typeof argsObj[1] !== 'function' ? argsObj[1] : null;
+      args.callback = !args.values ? argsObj[1] : (typeof argsObj[2] === 'function' ? argsObj[2] : undefined);
```

Comment:
Should the recent fix for `mysql` also apply here? https://github.com/aws/aws-xray-sdk-node/pull/66

Reply:
Hi, any update on this? Please let me know your thoughts.

Reply:
I updated it in my latest commit:  1474129

Reply:
sorry for the confusion. My question is that should the change `typeof argsObj[1] === 'function' ?` on `mysql_p.js` also be applied to `postgres_p.js` file here?

Reply:
Yes, I think it should.